### PR TITLE
[if] condition indentation change for Jane Street and an interesting OCaml parser edge case

### PR DESCRIPTION
**Pull request by pszilagyi**
Sorry, I know this is switching back and forth, but we changed our mind on the indentation for [if] conditions.  I hope it's easy to switch that one case back.

We also noticed an interesting edge case in the OCaml grammar for top-level [let] interacting with [function].  We added a test case to edge-cases.ml indicating how OCaml parses it, which is not at all intuitive.  (This is low priority for us.)

(We also removed one discouraged style from our tests, which is not important.)

Thanks for your help!

### CODE REVIEW

File: tests/failing/js-args.ml [hunk 1]
```diff
@@ -118,13 +118,18 @@ let _ =
...
+     expression and indent accordingly.  Whether [&&] or [||], conditionals
+     effectively state lists of conditions for [then]. *)
   if Edge_adjustment.is_zero arb.cfg.extra_edge
-     && 0. = sys.plugs.edge_backoff
-     && 0. = zero_acvol_edge_backoff
+  && 0. = sys.plugs.edge_backoff
+  && 0. = zero_acvol_edge_backoff
   then 0.
   else 1.
 let _ =
```
Comment (AltGr) [inline tests/failing/js-args.ml]: Should it be the same for `when` ? E.g.
```
let f = function
  | x when c1
        && c2 ->
    x
```

Comment (pszilagyi) [inline tests/failing/js-args.ml]: On Mon, Jul 27, 2015 at 9:04 PM, Louis Gesbert <notifications@github.com>
wrote:

> In tests/failing/js-args.ml
> <https://github.com/OCamlPro/ocp-indent/pull/185#discussion_r35605876>:
>
> >    if Edge_adjustment.is_zero arb.cfg.extra_edge
> > -     && 0. = sys.plugs.edge_backoff
> > -     && 0. = zero_acvol_edge_backoff
> > +  && 0. = sys.plugs.edge_backoff
> > +  && 0. = zero_acvol_edge_backoff
> >    then 0.
> >    else 1.
> >  let _ =
>
> Should it be the same for when ? E.g.
>
> let f = function
>   | x when c1
>         && c2 ->
>     x
>
> I think you have discretion in that case.  I can ask around but that's
just begging for trouble.  Obivously, we may change our mind in the future,
but I haven't heard any opinions about "when".

Comment (AltGr) [inline tests/failing/js-args.ml]: that was part of the same patch that I reverted and had already been around for a while, so I guess it's ok. If not it's very straightforward to change.
